Feature/create meeting pull request 요청

jihwankim129 opened the pull request:
흠 빌드같은거 해서 테스트 해보고 되는지 확인해서 해보고 싶엇는데 그냥 back-end꺼 브랜치 가져와서 실행시키면 계속 오류가 뜨더라구요. db 연결 로컬 mysql로도 돌리고 이것저것 해봣는데... 테스트 해보고 싶은데 자꾸 계속 안되서 일단 PR 을 남기고 

이거 기존에 작성했던거 이미 back-end 에 있는 예시에 맞춰서 수정을 해서 올렸습니다.

근데 브랜치 가져왔더니 lombok getter setter가 활성화가 왜안되는진 모르겟지만 계속 에러가 떠서 무튼

결론: 테스트는 제대로 못했지만 합쳐도 될지 한번 봐주세요

pileuszu committed:
feat: 모임 컨트롤러 생성

- MeetingController.java: 모임 컨트롤러 생성

pileuszu committed:
feat: 모임 컨트롤러에 필요한 dto 생성

- MeetingForm.java: 모임 생성 및 수정을 위한 dto
- ResignReasonForm.java: 모임 탈퇴/강퇴 사유를 위한 dto

pileuszu committed:
feat: 모임 컨트롤러와 모임 서비스 연결

- MeetingController.java: 모임 컨트롤러와 모임 서비스 연결
- MeetingService.java: 모임 서비스 추가

pileuszu committed:
feat: 모임 컨트롤러 JWT 추가

- MeetingController.java: JWT 추가

pileuszu committed:
feat: 모임 기능 Controller, Service, Repository 초기 구현

- Meeting.java: entity builder 에 필요한 기능 추가
- MeetingController.java: 컨트롤러 로직 구현
- MeetingService.java: 서비스 로직 구현
- MeetingRepository.java: 리포지토리 로직 구현
- MemberService.java: findMember 함수 생성
- ResignReasonForm.java: type 필드 생성 강퇴 / 탈퇴 구분

pileuszu committed:
feat: 불필요한 import 삭제

- MeetingController.java: 불필요한 import 삭제
- MeetingService.java: 불필요한 import 삭제
- MeetingRepository.java: 불필요한 import 삭제

pileuszu committed:
feat: 추가와 삭제 메소드시 Meeting과 Member 필드값도 삭제 필요

- MeetingController.java: 추가 필요
- MeetingService.java: 추가 필요
- MemberMeetingRepository.java: 추가 필요
- Member.java: 추가 필요

pileuszu committed:
feat: Meeting Service 함수 수정

- MeetingService.java: 함수 수정

pileuszu committed:
feat: Service단 dto제거 및 repository 내용 수정

- MeetingService.java: dto 제거 entity 사용
- MeetingRepository.java: 기존 함수들 제거후 JPARepository에 등록되더있는 query사용

pileuszu committed:
feat: MeetingController에 모임 관련 기능 추가

- Meeting.java: createMeeting 추가
- MeetingController.java: 모임 기능에 필요한 Mapping들 추가
- MeetingService.java: MeetingController에서 구현한 내용에 필요한 Service단 함수들 내용 추가

SyingSHY reviewed (changes requested):
일단 main으로 PR이 아니라 back-end로 PR입니다. 이 부분은 제가 직접 수정했습니다.

커밋 로그가 이렇게 남을 리가 없는데...?
그리고 pileuzsu 로컬 계정이신 거 같은데 git config user.email / user.name 확인 부탁드립니다.

여러모로 뭔가 잘못된 거 같은데 재검토 부탁 드립니다.

lee1684 commented:
9c06d46863a7bc28dc89158937c677fad0746b4a 이 커밋 하나만 추가하신 것 같은데 리베이스가 잘못된 것 같네요.
직접 하시다가 더 꼬일 수도 있어서 제가 푸쉬해놓겠습니다.

jihwankim129 commented:
내용이 많이 바껴서 리베이스보다 그냥 직접넣는게 더 빠른것 같아서 그냥 커밋 한개만 넣었어요.

머지 리퀘스트를 실수로 메인으로 했었군요

jihwankim129 commented:
계속 계정 왔다 갔다쓰다 보니깐 pileuszu 계정으로 들어갔군요

lee1684 commented on back-end/src/main/java/kr/co/ssalon/domain/service/MeetingService.java:
```diff
@@ -53,4 +59,90 @@ public Page<Meeting> getMoims(MeetingSearchCondition meetingSearchCondition, Pag
+        Member currentUser = memberService.findMember(username);
+
+        if (!meetingRepository.getReferenceById(moimId).getCreator().equals(currentUser)) {
+        }
```
이 부분 return이 누락된 걸까요?

jihwankim129 replied:
아하 저부분은 사실 예외 던질 곳인데 예외 어떻게 처리할지 안정해가지고, 일단 비워뒀는데 이미 작성하신거 보고 BadRequestException던지는걸로 바꿀게요

lee1684 commented on back-end/src/main/java/kr/co/ssalon/domain/service/MeetingService.java:
```diff
@@ -53,4 +59,90 @@ public Page<Meeting> getMoims(MeetingSearchCondition meetingSearchCondition, Pag
+
+        Meeting meeting = meetingRepository.getReferenceById(moimId);
+        if (meeting.getCreator().equals(currentUser)) {
+        }
```
이 부분도 !랑 return이 누락된 것 같네요.

jihwankim129 replied:
위에랑 동일한 사유입니다 대신 저건 조건 문 안에 not 연산 넣은다음 예외 던질게요

lee1684 commented on back-end/src/main/java/kr/co/ssalon/domain/service/MeetingService.java:
```diff
@@ -53,4 +59,90 @@ public Page<Meeting> getMoims(MeetingSearchCondition meetingSearchCondition, Pag
+
+        List<MemberMeeting> participants = meeting.getParticipants();
+        for (MemberMeeting memberMeeting : participants) {
+            Member roopMember = memberRepository.getReferenceById(memberMeeting.getMember().getId());
```
roopMember가 어떤 뜻일까요?

jihwankim129 replied:
멤버 엔티티에 있는 멤버미팅을 하나하나 지워주기 위해 쓴 코드고 각 멤버를 의미합니다.

lee1684 replied:
roop이 어떤 단어의 줄임말일까요?

jihwankim129 replied:
그냥 루프문으로 돌아다니면서 멤버에 접근하다 보니깐  말그대로 루프하다의 루프입니다

lee1684 replied:
아하 그럼 loop가 맞겠네요!

lee1684 replied:
그리고 밑에 코멘트를 못 보시고 푸쉬하셔서 커밋 로그가 다시 바뀌었는데 실습 시간에 다시 한 번 보죠.

jihwankim129 replied:
그러게요 loop로 바꿔야하네 일단 저거 바꿔놓을게요

lee1684 commented:
제가 force push 한 브랜치이기 때문에 지환님 로컬에 있는 feature/create-meeting 브랜치 지우시고 다시 git에 올라와 있는 브랜치 가져오셔야 합니다!

lee1684 commented:
병합 여부는 오늘 실습 시간에 테스트 해보고 나서 결정하죠.

jihwankim129 committed:
fix: MeetingService.java 예외 추가 및 deleteMoim 코드 수정

- MeetingService.java: 예외 추가 및 deleteMoim 코드 수정

jihwankim129 committed:
fix: member를 username으로 조회할때 optional을 반환해서 생기는 오류 해결

- MeetingService.java: 불러오는 함수 변경
- MemberRepository.java: 새 쿼리문 추가
- MemberService.java: 새로운 함수 추가

jihwankim129 committed:
fix: getter 오타 수정

- MeetingService.java: getter 오타 수정

jihwankim129 commented:
> 제가 force push 한 브랜치이기 때문에 지환님 로컬에 있는 feature/create-meeting 브랜치 지우시고 다시 git에 올라와 있는 브랜치 가져오셔야 합니다!

음 controller나 dto위치가 잘못된데 들어가있던 커밋이라 그냥 다 지워버렷엇는데 그러면 그냥 현재 브랜치에서 그런 내용들 지운거 커밋 추가할게요. 커밋이 뭔가 복구된것 같아서 잘못된 위치에 파일이 생겨버렷네요

jihwankim129 committed:
fix: 불필요한 파일 삭제

- MeetingController.java: 잘못된 위치에 있던 파일 삭제
- MeetingForm.java: 잘못된 위치에 있던 파일 삭제
- MemberMeetingService.java: 불필요한 내용 삭제
- ResignReasonForm.java: 잘못된 위치에 있던 파일 삭제

jihwankim129 commented:
> > 제가 force push 한 브랜치이기 때문에 지환님 로컬에 있는 feature/create-meeting 브랜치 지우시고 다시 git에 올라와 있는 브랜치 가져오셔야 합니다!
> 
> 음 controller나 dto위치가 잘못된데 들어가있던 커밋이라 그냥 다 지워버렷엇는데 그러면 그냥 현재 브랜치에서 그런 내용들 지운거 커밋 추가할게요. 커밋이 뭔가 복구된것 같아서 잘못된 위치에 파일이 생겨버렷네요

미쳐 발견못하고 지운게 있을까봐 걱정되긴 한데 일단 눈에 보이는건 지워놨습니다. 막 controller위치가 web 디렉토리 아래에 있지 않다던가 불필요한 코드가 추가되어있다던가 한 내용들인데 눈에 보이는건 지워놨습니다.

jihwankim129 commented:
다른 코드들 보니깐 /api 가 추가되어있던데 저도 그런식으로 바꿀까요

jihwankim129 commented:
리베이스를 안하고 그냥 있던 내용들 다 지우고 새로 create-meeting 을 백엔드에서 가져와서 그거 기준으로 코드 다시 처음부터 짠거였어요

jihwankim129 committed:
fix: 중복된 함수명으로 생긴 오류 해결

- MemberService.java: findMember 중복 제거

jihwankim129 committed:
fix: MeetingDto내부 속성 타입이 변경됨에 따라 코드 수정 및 Member와 MemberMeeting Meeting 간의 양방향 매핑에서 발생하는 추가 부분 코드 수정

- Meeting.java: 양방향 매핑을 위한 함수 추가
- MeetingService.java: 양방향 매핑을 위한 코드 수정
- Member.java: 양방향을 위한 cascade 추가 및 함수 추가
- MemberMeeting.java: 양방향 매핑을 위한 builder 내용 수정
- MemberMeetingService.java: null 값을 잡아내기 위해 getReference를 지우고 validation해서 exception을 던지는 코드 추가
- MemberRepository.java: 기존 존재하던 findMember를 활용하기 위해 새로 생성했던 쿼리문 다시 삭제
- MemberService.java: id를 통해 member를 서치하고 validation하는 함수 추가
- PaymentRepository.java: payment DB 조회를 위한 리포지토리 추가 JpaRepository 상속
- TicketRepository.java: Ticket DB 조회를 위한 리포지토리 추가 JpaRepository 상속

jihwankim129 commented:
meetingDTO가 id를 쓰도록 바뀌어 있더군요 그거에 맞춰서 한번 수정해보았습니다.
